File: 1. Binary Search/BinarySearch.java
import java.util.*;

public class BinarySearch {
	public static boolean binarySearch(int[] arr, int target) {
		return binarySearchHelper(arr, target, 0, arr.length - 1);
	}

	private static boolean binarySearchHelper(int[] arr, int target, int low, int high) {
		if(low > high) return false;
		int mid = (int)(low + high)/2;
		if(arr[mid] == target) {
			return true;
		}
		else if(arr[mid] > target) return binarySearchHelper(arr, target, low, mid - 1);
		else return binarySearchHelper(arr, target, mid+1, high);
	}

	public static void main(String[] args) {
		int[] arr = {0, 1, 2, 3, 4, 5, 6, 7};

		assert binarySearch(arr, 2): "failed at checking 2";
		assert binarySearch(arr, 7): "failed at checking 7";
		assert (binarySearch(arr, 10) == false) : "failed at checking 10";
		assert (binarySearch(arr, 0)) : "failed at checking 0";
	}
}

File: 1. Binary Search/binary_search.js
'using strict';

function binarySearch(arr, target) {
	if(arr.length === 0) return false;

	let low = 0, high = arr.length - 1;
	while(low <= high) {
		let mid = Math.round((low + high) / 2);
		if(arr[mid] === target) return true;
		else if(arr[mid] > target) high = mid - 1;
		else low = mid	+ 1;
	}
	return false;
}

let arr = [1, 2, 3, 4, 5];
console.log(binarySearch(arr, 1));